src/main/java/net/earthcomputer/clientcommands/features/PlayerRandCracker.java

```diff
@@ -250,29 +271,40 @@ private static boolean canMaintainPlayerRNG() {
         }
     }
 
-
     // ===== UTILITIES ===== //
-
     public static boolean throwItemsUntil(Predicate<Random> condition, int max) {
-        if (!TempRules.playerCrackState.knowsSeed())
+        return throwItemsUntil(condition, max, false);
+    }
+
+    public static boolean throwItemsUntil(Predicate<Random> condition, int max, boolean sendErrors) {
+        if (!TempRules.playerCrackState.knowsSeed()) {
             return false;
+        }
         TempRules.playerCrackState = CrackState.CRACKED;
 
         long seed = PlayerRandCracker.seed;
         Random rand = new Random(seed ^ MULTIPLIER);
 
         int itemsNeeded = 0;
         for (; itemsNeeded <= max && !condition.test(rand); itemsNeeded++) {
-            for (int i = 0; i < 4; i++)
+            for (int i = 0; i < 4; i++) {
                 seed = (seed * MULTIPLIER + ADDEND) & MASK;
+            }
             rand.setSeed(seed ^ MULTIPLIER);
         }
-        if (itemsNeeded > max)
+        if (itemsNeeded > max) {
+            if (sendErrors) {
+                sendError(new TranslatableText("playerManip.throwError", itemsNeeded));
+            }
             return false;
-
+        }
         for (int i = 0; i < itemsNeeded; i++) {
-            if (!throwItem())
+            if (!throwItem()) {
+                if (sendErrors) {
+                    sendError(new TranslatableText("playerManip.notEnoughItems", i, itemsNeeded));
+                }
```

Comment:
Why do you need these errors now if you are already emitting an error in `ChorusManipulation.onEat`?

Reply:
For detailed feedback.
There can be two reasons a teleport fails. And I think for the end user it is helpfull, to know why he isn't able to manipulate his location.

Reply:
Make an enum with an "OK" result and other values for possible types of failures. Something similar is done for enchanting manipulation.

Reply:
So, throwItemUntil should return an Enum?

Reply:
Yes